File: src/main/java/com/scherule/calendaring/api/verticles/MeetingApi.java
package com.scherule.calendaring.api.verticles;

import com.scherule.calendaring.domain.Meeting;
import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;

public interface MeetingApi {

    void createMeeting(Meeting meeting, Handler<AsyncResult<Meeting>> handler);

    void getMeeting(String meetingId, Handler<AsyncResult<Meeting>> handler);

    void removeMeeting(Long meetingId, Handler<AsyncResult<Void>> handler);

    void updateMeeting(Meeting meeting, Handler<AsyncResult<Meeting>> handler);

}

File: src/main/kotlin/com.scherule.calendaring/api/verticles/MeetingApi.kt
package com.scherule.calendaring.api.verticles

import com.scherule.calendaring.domain.Meeting
import rx.Completable
import rx.Single

interface MeetingApi {

    fun createMeeting(meeting: Meeting): Single<Meeting>

    fun getMeeting(meetingId: String): Single<Meeting>

    fun removeMeeting(meetingId: Long): Completable

    fun updateMeeting(meeting: Meeting): Single<Meeting>

}
